### DISCUSSION TITLE
PI ODBC Required?

### DISCUSSION THREAD

**Discussion by rhoech** [Hugovdberg/PIconnect#644] [Q&A]
The PI Server I am accessing does not have PI Integrator set up 

https://docs.osisoft.com/bundle/pi-integrator-for-business-analytics/page/pi-integrator-for-business-analytics.html

I had to install PI ODBC driver on my local computer to access the database server ...

With this python package, can you access PI Databases without the use of PI ODBC?  From a remote computer also?  Or just on the same computer where the PI Server resides?

Also, has this been tested on Linux also, or just Windows?

Thank you for your support.

Sincerely,
-Raita

**Comment by Hugovdberg** [comment]
The PI ODBC driver is not a requirement for this package. The only requirement is a working installation of the PI AF SDK, typically installed in "C:\Program Files (x86)\PIPC\AF\PublicAssemblies\4.0". It does not need to be run from the same machine as the PI server is running on, only the client components are necessary.

I have done a little testing of the SDK on Ubuntu, but it appears the authentication mechanism relies on Windows only protocols. It is possible that this will become easier if the SDK is translated to .NET Core, but is uncertain if or when that would happen: https://feedback.osisoft.com/forums/555145-pi-developer-technologies/suggestions/40291651-add-net-core-support-for-af-sdk?page=1&per_page=20
